girder/utility/s3_assetstore_adapter.py

```diff
@@ -117,6 +125,18 @@ def _getRequestHeaders(self, upload):
             'x-amz-meta-uploader-ip': str(cherrypy.request.remote.ip)
         }
 
+    def _generatePresignedUrl(self, *args, **kwargs):
+        """
+        Wrap self.client.generate_presigned_url to allow it work with
+        Google Cloud Storage.
+
+        See https://gist.github.com/gleicon/2b8acb9f9c0f22753eaac227ff997b34
+        """
+        url = self.client.generate_presigned_url(*args, **kwargs)
+        if getattr(self.client, '_useGoogleAccessId', False):
+            url = url.replace('AWSAccessKeyId', 'GoogleAccessId')
```

Comment:
Sorry for the confusion, this is actually where I was talking about using `urlparse` rather than `str.replace`, though it's also an improvement where you changed it up above.